src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -401,11 +401,11 @@ private void innerCreateNoLock(Create create, IndexWriter writer, long currentVe
         if ((versionValue != null && versionValue.delete() == false) || (versionValue == null && currentVersion != Versions.NOT_FOUND)) {
             if (create.origin() == Operation.Origin.RECOVERY) {
                 return;
-            } else if (create.origin() == Operation.Origin.REPLICA) {
+            } else if (create.origin() == Operation.Origin.REPLICA && !create.autoGeneratedId()) {
```

Comment:
I think this is wrong? I this is an indexing request on a replica and we're here, that means that there is already a doc with this id. In this case we want to just ignore the request.  +1 on what Simon said regarding not changing this code.